### PR TITLE
Rename WithContainerRunArgs to WithContainerRuntimeArgs to reduce potential confusion

### PR DESCRIPTION
**Pull request by danegsta**
The original API name was confusing users (see #3944); renaming to try to reduce confusion since this is intended to be an advanced scenario.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3950)

### CODE REVIEW

File: src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs [hunk 1]
```diff
@@ -177,9 +177,9 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
     /// <param name="builder">The resource builder.</param>
     /// <param name="args">The arguments to be passed to the container run command when the container resource is started.</param>
```
Comment (DamianEdwards) [inline src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs:178]:
```suggestion
    /// <param name="args">The arguments to be passed to the container host run command when the container resource is started.</param>
```

File: src/Aspire.Hosting/ApplicationModel/ContainerHostRunArgsCallbackAnnotation.cs [hunk 2]
```diff
@@ -9,24 +9,24 @@ namespace Aspire.Hosting.ApplicationModel;
 /// Represents an additional argument to pass to the container run command.
```
Comment (DamianEdwards) [inline src/Aspire.Hosting/ApplicationModel/ContainerHostRunArgsCallbackAnnotation.cs:9]:
```suggestion
/// Represents an additional argument to pass to the container host run command.
```

File: src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs [hunk 3]
```diff
@@ -205,9 +205,9 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
     /// <param name="builder">The resource builder.</param>
```
Comment (DamianEdwards) [inline src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs:205]: Can we update the doc comment on this one too?

File: src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs [hunk 4]
```diff
@@ -177,9 +177,9 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
...
     /// <param name="args">The arguments to be passed to the container run command when the container resource is started.</param>
     /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<T> WithContainerRunArgs<T>(this IResourceBuilder<T> builder, params string[] args) where T : ContainerResource
+    public static IResourceBuilder<T> WithContainerHostRunArgs<T>(this IResourceBuilder<T> builder, params string[] args) where T : ContainerResource
```
Comment (eerhardt) [inline src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs:180]: (minor) is there something we can add/change/enhance on the XML docs to help callers understand the difference between this and WithArgs?

Maybe we can add a `<remarks>` section?

Comment (danegsta) [reply src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs:180]: Does anyone know the cref syntax for a function with a `params` argument?

Comment (danegsta) [reply src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs:183]: I think I've got the right format

File: src/Aspire.Hosting/Dcp/ApplicationExecutor.cs [hunk 5]
```diff
@@ -1600,7 +1600,7 @@ private void AddServicesProducedInfo(IResource modelResource, IAnnotationHolder
             }
             else
             {
-                
+
```
Comment (eerhardt) [inline src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:1603]: (nit) can we either revert the whitespace change, or remove this empty line?

File: src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs [hunk 6]
```diff
@@ -171,43 +171,52 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
     }
 
     /// <summary>
-    /// Adds a callback to be executed with a list of arguments to add to the container run command when a container resource is started.
+    /// Adds a callback to be executed with a list of arguments to add to the container host run command when a container resource is started.
     /// </summary>
+    /// <remarks>
+    /// This is intended to pass additional arguments to the underlying container run command to enable advanced features such as exposing GPUs to the container. To pass runtime arguments to the actual container, use the <see cref="ResourceBuilderExtensions.WithArgs{T}(IResourceBuilder{T}, string[])"/> annotation.
```
Comment (eerhardt) [inline src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs:177]:
```suggestion
    /// This is intended to pass additional arguments to the underlying container run command to enable advanced features such as exposing GPUs to the container. To pass runtime arguments to the actual container, use the <see cref="ResourceBuilderExtensions.WithArgs{T}(IResourceBuilder{T}, string[])"/> method.
```

In all 3 places.

**Review by eerhardt — APPROVED** [review]
This looks good to me. Thanks for the fix.

Be sure all the changes get to the release/8.0 PR.

**Pull request merged**